service/pushbullet/pushbullet.go

```diff
@@ -0,0 +1,67 @@
+package pushbullet
+
+import (
+	"github.com/cschomburg/go-pushbullet"
+	"github.com/pkg/errors"
+)
+
+// Pushbullet struct holds necessary data to communicate with the Pushbullet API.
+type Pushbullet struct {
+	client          *pushbullet.Client
+	deviceNicknames []string
+}
+
+// New returns a new instance of a Pushbullet notification service.
+// For more information about Pushbullet api token:
+//    -> https://docs.pushbullet.com/#api-overview
+func New(apiToken string) *Pushbullet {
+	client := pushbullet.New(apiToken)
+
+	pb := &Pushbullet{
+		client:          client,
+		deviceNicknames: []string{},
+	}
+
+	return pb
+}
+
+// AddReceivers takes Pushbullet device nicknames and adds them to the internal deviceNicknames list. 
+// The Send method will send a given message to all those devices a matching one be registered.
+// We only add registered devices.
+func (pb *Pushbullet) AddReceivers(deviceNicknames ...string) {
+
+	devices := []string{}
+	for _, deviceNickname := range deviceNicknames {
+
+		_, err := pb.client.Device(deviceNickname)
+		if err != nil {
+			continue
+		}
+		devices = append(devices, deviceNickname)
+	}
+	pb.deviceNicknames = append(pb.deviceNicknames, deviceNicknames...)
```

Comment:
Not too sure about this part. We're basically testing twice if a device is registered - here and in _line 53_ again.

_AddReceivers_ is not a validation function and as its name suggests it should only add new receivers to the list. And as you say in the comment below, devices may get deregistered during runtime - that means that we have to check the devices everytime we call the _Send_ method anyway. 

I personally have never used Pushbullet before but if I understand you correctly, then the registration check inside the _AddReceivers_ function should be unnecessary. 

But please, I don't know Pushbullet at all, correct me if I'm wrong. Good work nonetheless and really appreciate your contributions. 

In case my understanding is correct, please apply:

```suggestion
	pb.deviceNicknames = append(pb.deviceNicknames, deviceNicknames...)
```

Reply:
I suppose the semantic difference here is the behaviour on failure.
Here AddReceivers simply ignores invalid devices, in Send it would return an error on an invalid device.
What would the prefered behaviour be for you? In my mind you don't want to stop sending to everything if only a single device is invalid, hence this double check.

For reference, one would have to actively deregister a device in between adding the receiver and sending the message - it's a manual process.

Happy to go with whatever you feel is more sensible

Reply:
Update: Adjusted it to simply continue on an unregistered device. I'd be interested to hear any thoughts on this - a similar case may be sending emails to an invalid address - is the onus on the user?